server/server.go

```diff
@@ -9,17 +9,67 @@ import (
 
 	"github.com/bombsimon/amqp-rpc/connection"
 	"github.com/bombsimon/amqp-rpc/logger"
-	"github.com/bombsimon/amqp-rpc/middleware"
+)
+
+type ctxKey string
+
+const (
+	// CtxQueueName can be used to get the queue name from the context.Context
+	// inside the HandlerFunc.
+	CtxQueueName ctxKey = "queue_name"
 )
 
 var (
 	// ErrResponseChClosed is an error representing a closed response channel.
 	ErrResponseChClosed = errors.New("Channel closed")
 )
 
-// handlerFunc is the function that handles all request based on the routing
+// HandlerFunc is the function that handles all request based on the routing
 // key.
-type handlerFunc func(context.Context, amqp.Delivery) []byte
+type HandlerFunc func(context.Context, *ResponseWriter, amqp.Delivery)
+
+// ResponseWriter is used by a handler to construct an RPC response.
+// The ResponseWriter may NOT be used after the handler has returned.
+type ResponseWriter struct {
```

Comment:
Maybe the response writer and it's methods should live in it's own file (i.e `response_writer.go`)? I can see multiple new methods in a near future for this! 

I also think we should keep this file as simple as possible.

Reply:
Sure, we can create a `response_writer.go`, I like that.